resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsSnapshot.scala

```diff
@@ -72,5 +74,8 @@ object ExecutorPodsSnapshot extends Logging {
     }
   }
 
-  private def isDeleted(pod: Pod): Boolean = pod.getMetadata.getDeletionTimestamp != null
+  private def isDeleted(pod: Pod): Boolean = {
+    (pod.getMetadata.getDeletionTimestamp != null &&
+     pod.getStatus.getPhase.toLowerCase(Locale.ROOT) != "terminating")
```

Comment:
This seems to cause NPE.
```
sbt.ForkMain$ForkError: java.lang.NullPointerException: null	at org.apache.spark.scheduler.cluster.k8s.ExecutorPodsSnapshot$.isDeleted(ExecutorPodsSnapshot.scala:79)
```

Reply:
Weird, it's passing locally I don't know why this would fail in Jenkins. I'll do some digging.

Reply:
Wait where is this from? I don't see that in the K8s Jenkins logs?

Reply:
Oops. I missed your comment here. It was in UT failure (instead of K8s IT).